worldedit-core/src/main/java/com/sk89q/worldedit/world/World.java

```diff
@@ -107,7 +110,30 @@
      * @param notifyAndLight true to to notify and light
      * @return true if the block was successfully set (return value may not be accurate)
      */
-    <B extends BlockStateHolder<B>> boolean setBlock(BlockVector3 position, B block, boolean notifyAndLight) throws WorldEditException;
+    @Deprecated
+    default <B extends BlockStateHolder<B>> boolean setBlock(BlockVector3 position, B block, boolean notifyAndLight) throws WorldEditException {
+        return setBlock(position, block, notifyAndLight ? SideEffectSet.defaults() : SideEffectSet.none());
+    }
+
+    /**
+     * Similar to {@link Extent#setBlock(BlockVector3, BlockStateHolder)} but a
+     * {@code notifyAndLight} parameter indicates whether adjacent blocks
+     * should be notified that changes have been made and lighting operations
+     * should be executed.
+     *
+     * <p>If it's not possible to skip lighting, or if it's not possible to
+     * avoid notifying adjacent blocks, then attempt to meet the
+     * specification as best as possible.</p>
+     *
+     * <p>On implementations where the world is not simulated, the
+     * {@code notifyAndLight} parameter has no effect either way.</p>
+     *
+     * @param position position of the block
+     * @param block block to set
+     * @param sideEffectSet which side effects to perform
+     * @return true if the block was successfully set (return value may not be accurate)
+     */
+    <B extends BlockStateHolder<B>> boolean setBlock(BlockVector3 position, B block, SideEffectSet sideEffectSet) throws WorldEditException;
```

Comment:
Do we want to report the same information as `applySideEffects` return value somehow?

Reply:
I personally think for this one it's not as necessary. The point of this method is to set the block, so the return type should reflect whether that was done correctly. The side effects are just side effects, after all.